core/src/vdaf.rs

```diff
@@ -607,288 +600,6 @@ macro_rules! vdaf_dispatch {
-    fn validate(value: &str) -> Result<(), anyhow::Error> {
-        let mut iter = value.chars();
-        let mut any_non_equals = false;
-        // First loop: consume "normal" characters, stop when we see an equals sign for padding or
-        // reach the end of the input.
-        for c in &mut iter {
-            match c {
-                'a'..='z' | 'A'..='Z' | '0'..='9' | '-' | '.' | '_' | '~' | '+' | '/' => {
-                    any_non_equals = true;
-                }
-                '=' => {
-                    if !any_non_equals {
-                        return Err(anyhow::anyhow!("bearer token may not start with '='"));
-                    }
-                    break;
-                }
-                _ => return Err(anyhow::anyhow!("bearer token may not contain '{c}'")),
-            }
-        }
-        // Second loop: consume any further padding characters, if present.
-        for c in &mut iter {
-            match c {
-                '=' => {}
-                _ => {
-                    return Err(anyhow::anyhow!(
-                        "bearer token may only contain '=' at the end"
-                    ))
-                }
-            }
-        }
-        Ok(())
-    }
-}
-
-impl AsRef<str> for BearerToken {
-    fn as_ref(&self) -> &str {
-        &self.0
-    }
-}
-
-impl AsRef<[u8]> for BearerToken {
-    fn as_ref(&self) -> &[u8] {
-        self.0.as_bytes()
-    }
-}
-
-impl TryFrom<String> for BearerToken {
-    type Error = anyhow::Error;
-
-    fn try_from(value: String) -> Result<Self, Self::Error> {
-        Self::validate(&value)?;
-        Ok(Self(value))
-    }
-}
-
-impl TryFrom<Vec<u8>> for BearerToken {
-    type Error = anyhow::Error;
-
-    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
-        Self::try_from(String::from_utf8(value)?)
-    }
-}
-
-impl<'de> Deserialize<'de> for BearerToken {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        String::deserialize(deserializer)
-            .and_then(|string| Self::try_from(string).map_err(D::Error::custom))
-    }
-}
-
-impl PartialEq for BearerToken {
-    fn eq(&self, other: &Self) -> bool {
-        // We attempt constant-time comparisons of the token data to mitigate timing attacks. Note
-        // that this function still leaks whether the lengths of the tokens are equal -- this is
-        // acceptable because we expect the content of the tokens to provide enough randomness that
-        // needs to be guessed even if the length is known.
-        constant_time::verify_slices_are_equal(self.0.as_bytes(), other.0.as_bytes()).is_ok()
-    }
-}
-
-impl Eq for BearerToken {}
-
-impl Distribution<BearerToken> for Standard {
-    fn sample<R: rand::Rng + ?Sized>(&self, rng: &mut R) -> BearerToken {
-        BearerToken(URL_SAFE_NO_PAD.encode(rng.gen::<[u8; 16]>()))
-    }
-}
-
 /// Returns the given [`Url`], possibly modified to end with a slash.
```

Comment:
I think we should hoist this to the top level of the crate, since it's not related to VDAFs at all.